crates/rb-sys/src/stable_api.rs

```diff
@@ -54,6 +54,22 @@ pub trait StableApiDefinition {
     /// is valid.
     unsafe fn rarray_const_ptr(&self, obj: VALUE) -> *const VALUE;
 
+    /// Get the class from a VALUE which contains an RBasic struct.
+    ///
+    /// # Safety
+    /// This function is unsafe because it dereferences a raw pointer to get
+    /// access to underlying RBasic struct. The caller must ensure that the
+    /// `VALUE` is a valid pointer to a non-immediate object.
+    unsafe fn rbasic_class(&self, obj: VALUE) -> VALUE;
```

Comment:
I'm wondering if we should handle the null case more explicitly (hidden class), so make this:

```suggestion
    unsafe fn rbasic_class(&self, obj: VALUE) -> Option<NonNull<VALUE>>;
```

Reply:
Worth trying, that could be a benefit of the stable API: to leverage Rust type system when it makes sense to do so. Will give it a shot!